### PR TITLE
Assigning a fix Version to a set of issues.

### PR DESCRIPTION
**Pull request by dazinator**
This PR is to add an action to the command line, that allows assigning a fix Version to a specified set of issues (for #4)

The fix version must already be created, and the issues must already exist.

The command line is:

`[basecall] -action AssignVersion -project "KEY" -version 0.0.1 -issues "KEY-1,KEY-2,KEY-3"`

The update only appends the specified version to the fix version field on the issues, it won't overwrite or remove existing fix versions in the field.

### CODE REVIEW

**Comment by dazinator** [comment]
In reality I'll be wanting to get the list of issues from Team City for the current build, however I've assumed you want to keep JiraCli completely agnostic of build systems, so I'll put all that Team City logic in a seperate tool for now.

**Comment by GeertvanHorrik** [comment]
Good assumption. We don't use (or plan to use) teamcity. I think it's best to pass them to the tool instead. Will review soon.

File: src/JiraCli.Tests/IntegrationTests/IntegrationTests.cs [hunk 1]
```diff
@@ -26,15 +27,16 @@ private T ResolveService<T>()
...
         {
             var context = new Context
             {
                 JiraUrl = JiraUrl,
                 Project = JiraProject,
                 UserName = JiraUser,
                 Password = JiraPassword,
-                Version = version
+                Version = version,
+                Issues = issues
```
Comment (GeertvanHorrik) [inline src/JiraCli.Tests/IntegrationTests/IntegrationTests.cs]: Maybe best to use this:

```
issues ?? new string[] { }
```

That way we never have to check for null on issues.

File: src/JiraCli/ArgumentParser.cs [hunk 2]
```diff
@@ -107,6 +107,12 @@ public static Context ParseArguments(List<string> commandLineArguments)
                     continue;
                 }
 
+                if (IsSwitch("issues", name))
+                {
+                    context.Issues = value.Split(new char[] { ',' }, StringSplitOptions.RemoveEmptyEntries);
```
Comment (GeertvanHorrik) [inline src/JiraCli/ArgumentParser.cs]: I suggest we do a Trim() here as well after splitting.

File: src/JiraCli.Tests/IntegrationTests/IntegrationTests.cs [hunk 3]
```diff
@@ -77,5 +79,20 @@ private Context CreateContext(string version)
             var createStableVersionContext = CreateContext("2.0.0");
             Assert.IsTrue(await action.Execute(createStableVersionContext));
         }
+
+        
+        [TestCase("TS-1,TS-2,TS-3")]
```
Comment (GeertvanHorrik) [inline src/JiraCli.Tests/IntegrationTests/IntegrationTests.cs:84]: Please add this test case as well:

[TestCase("TS-1, TS-2, TS-3")] 

It'll probably show that we should use Trim()

File: src/JiraCli/Services/VersionService.cs [hunk 4]
```diff
@@ -38,6 +39,38 @@ public VersionService(IMergeVersionService mergeVersionService, IVersionInfoServ
...
+            Argument.IsNotNullOrWhitespace(() => projectKey);
+            Argument.IsNotNullOrWhitespace(() => version);
+            Argument.IsNotNullOrEmptyArray(() => issues);
+
+            Log.Debug("Checking if version already exists");
+
+            var existingVersion = GetProjectVersion(jiraRestClient, projectKey, version);
+            if (existingVersion == null)
+            {
+                var error = string.Format("Version '{0}' does not exist/ Ensure you have created this version first.", version);
```
Comment (GeertvanHorrik) [inline src/JiraCli/Services/VersionService.cs]: Please use $"" instead.

I think we can even replace this call as a whole:

```
Log.ErrorAndThrowException<InvalidOperationException>($"Version '{version}' does not exist / ensure you have created this version first")
```

File: src/JiraCli/Services/VersionService.cs [hunk 5]
```diff
@@ -38,6 +39,38 @@ public VersionService(IMergeVersionService mergeVersionService, IVersionInfoServ
...
+                Log.Error(error);
+                throw new InvalidOperationException(error);
+            }
+
+            Log.Info("Version '{0}' exists", version);
```
Comment (GeertvanHorrik) [inline src/JiraCli/Services/VersionService.cs]:
```
Log.Info("Version '{version}' exists, going to set this as FixVersion for '{issues.Length}' issues"); 
```

**Comment by GeertvanHorrik** [comment]
PR looks really good, thanks for keeping the coding standards 👍 

If you want, you can add yourself to the CONTRIBUTORS file as well.

File: src/JiraCli.Tests/IntegrationTests/IntegrationTests.cs (hunk 1 above)
Comment (dazinator) [inline src/JiraCli.Tests/IntegrationTests/IntegrationTests.cs]: Good idea

File: src/JiraCli/ArgumentParser.cs (hunk 2 above)
Comment (dazinator) [inline src/JiraCli/ArgumentParser.cs]: Yup good catch

File: src/JiraCli.Tests/IntegrationTests/IntegrationTests.cs (hunk 3 above)
Comment (dazinator) [inline src/JiraCli.Tests/IntegrationTests/IntegrationTests.cs:84]: Yep - there is no test coverage for the actual argument parser dealing with this though. I'll see if I can add some

File: src/JiraCli/Services/VersionService.cs (hunk 4 above)
Comment (dazinator) [inline src/JiraCli/Services/VersionService.cs]: Ok

File: src/JiraCli/ArgumentParser.cs (hunk 2 above)
Comment (dazinator) [inline src/JiraCli/ArgumentParser.cs]: It appears that the argument parser doesn't support spaces being used within argument values. It splits out argument values based on space:

```
  public static Context ParseArguments(string commandLineArguments)
        {
            return ParseArguments(commandLineArguments.Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries).ToList());
        }
```

So using an argument value like: "test-1, test-2" would throw a spanner in the works, where as "test-1,test-2" is fine.

Although I think this particular method is only called during tests - i'll see If I can make it a little bit more robust.

**Comment by dazinator** [comment]
Cool - I've been through and hopefully addressed all the things you have raised! 
Wahoo contributor fame! :-)

**Comment by GeertvanHorrik** [comment]
Sorry for late reply, have been a bit busy with other things this week. Looks really good.

**Pull request merged**